src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,50 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = findIndex(key);
+        if (index != -1) {
+            values[index] = value;
+        } else if (size < MAX_SIZE) {
+            keys[size] = key;
+            values[size++] = value;
+        } else {
+            throw new RuntimeException("Storage is full");
+        }
```

Comment:
In general the less `else` conditions we have the more easy to read code is
```suggestion
        int index = findIndex(key);
        if (index != -1) {
            values[index] = value;
            return;
        } 
        if (size < MAX_SIZE) {
            throw new RuntimeException("Storage is full");
        } 
        keys[size] = key;
        values[size++] = value;
```